backend/app/services/knowledge/orchestrator.py

```
@@ -0,0 +1,900 @@
+                # Parse HTML and extract text
+                soup = BeautifulSoup(response.text, "html.parser")
+                # Remove script and style elements
+                for element in soup(["script", "style", "nav", "footer", "header"]):
+                    element.decompose()
+                text = soup.get_text(separator="\n", strip=True)
+                return text.encode("utf-8"), "txt"
+            else:
+                # Return raw content
+                return response.content, "txt"
+
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)